include/aspect/initial_conditions/interface.h

```diff
@@ -93,6 +94,31 @@ namespace aspect
     };
 
 
+    /**
+     * Create a class inheriting from dealii.Function to translate the
+     * interface call to dealii.Function format
+     */
+    template <int dim>
+    class InterfaceFunction : public dealii::Function<dim, double>
+    {
+      public:
+        InterfaceFunction (const Interface<dim> &interface,
+                           const unsigned int selected_component,
+                           const unsigned int n_components);
```

Comment:
`std::bind` will call a virtual function when you pass the address of the virtual function and an object of derived type. Witness the magic of C++:
```
#include <iostream>
#include <functional>

class B
{
public:
  virtual void f() = 0;
};

class D : public B 
{
public:
  virtual void f() { std::cout << "YES!" << std::endl; }
};


int main ()
{
  D d;
  auto x = std::bind (&B::f, std::ref(d));
  x();
}
```